src/os/source/vectorsource.js

```diff
@@ -987,6 +987,55 @@ os.source.Vector.prototype.setColumnAutoDetectLimit = function(value) {
 };
 
 
+/**
+ * Sets the colors of the provided features
+ *
+ * @param {Array<!ol.Feature>|null} items
+ * @param {string|null} color
+ *
+ * @suppress {accessControls}
+ */
+os.source.Vector.prototype.setColor = function(items, color) {
+  if (!items || !color) return;
+
+  var cm = this.colorModel;
+  if (cm) {
+    cm.colorFeatures(items, color);
+  } else {
+    // TODO figure out how to do the binning when there are no bins...
+    for (const item of items) {
+      var oldColor = os.color.toHexString(os.feature.getColor(item, undefined, undefined));
```

Comment:
- I would expect this function to always set `os.data.RecordField.COLOR` on the feature, so even if that matches the current feature color (ie, layer color or feature style config) the override will remain if the layer/feature style are changed.
- Using `os.feature.getColor` is great to avoid unnecessary events, but isn't a very performant function to call on a lot of features. It may be overkill given my previous point.
- Passing `undefined` to a function is superfluous if you aren't providing later arguments, because `undefined` is the default value when a parameter isn't provided.

Reply:
I would also add that calls to `os.color.toHexString` are slow. We should do extensive performance testing with large datasets with these changes.

Reply:
Sounds good.  This is the same way that the ColorModel colors features, so whatever improvements we make here should also be applied there and anywhere else that modifies a lot of feature colors at once -- likely making all of them call a new utility function in the Source.

Also, I needed the old color in order to fire the COLOR featureEvent -- which I think is what the UI uses to recolor the Feature in cesium.   Is it okay to just pretend oldColor was null?  (I'll make the change see what happens locally, but even if it works, I'm wondering if it's an OK thing to do)

Reply:
Here we go -- rather than use the nice getColor() method that goes and looks everywhere, I just do a raw values_[feature....COLOR] comparison.  Best of both worlds.  Also, I removed the toHexString() call and made it so the passed-in color is an RGBA instead of hex so the conversion is a one-time thing instead of a million-time thing.